modules/social_features/social_group/modules/social_group_request/social_group_request.module

```diff
@@ -128,96 +142,37 @@ function social_group_request_theme_registry_alter(&$theme_registry) {
 }
 
 /**
- * Implements hook_preprocess_group().
+ * Implements hook_preprocess_HOOK().
  */
-function social_group_request_preprocess_group(&$variables) {
-  /** @var \Drupal\group\Entity\GroupInterface $group */
-  $group = $variables['group'];
-  $group_type = $group->getGroupType();
-
-  $account = \Drupal::currentUser();
-  // If the user is already a member we don't bother processing any further.
-  if ($group->getMember($account)) {
+function social_group_request_preprocess_group(array &$variables): void {
+  if (!\Drupal::config('social_group.settings')->get('social_group_type_required')) {
     return;
   }
 
-  if (!$group_type->hasContentPlugin('group_membership_request')) {
-    return;
-  }
-
-  // If user has a pending invite we should skip the request button.
-  if (\Drupal::hasService('ginvite.invitation_loader')) {
-    /** @var \Drupal\ginvite\GroupInvitationLoader $loader */
-    $loader = \Drupal::service('ginvite.invitation_loader');
-    $group_invites = count($loader->loadByProperties(['gid' => $group->id(), 'uid' => $account->id()]));
-    if (NULL !== $group_invites && $group_invites > 0) {
-      return;
-    }
-  }
-
-  $group_types = ['flexible_group'];
-  \Drupal::moduleHandler()->alter('social_group_request', $group_types);
```

Comment:
This tag means that the deprecated hook should be available but all use-cases should use now a new hook that replaced the deprecated. Also, since almost all logic moved to the "Social Group" module so code that was here can be simply removed.

Reply:
I do understand the meaning of the tag, but, this hook is no longer invoked by our code ( or I could not find a place where the hook is invoked) , therefore all implementations of it will no longer be executed, making this a BC breaking change.